From e56e593b4feb7ea753646a8cf8b44dce0b16f72f Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 27 Sep 2023 14:39:51 +0200
Subject: [PATCH] arm: mach-k3: am642_init: support spl_boot_device in u-boot
 proper

The memory addresses for boot mode are still valid in full u-boot.
Compile the spl_boot_device function unconditionally so that it can be
available any time.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm/mach-k3/am642_init.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-k3/am642_init.c b/arch/arm/mach-k3/am642_init.c
index 4b95e81e00..593584115d 100644
--- a/arch/arm/mach-k3/am642_init.c
+++ b/arch/arm/mach-k3/am642_init.c
@@ -331,6 +331,7 @@ u32 spl_mmc_boot_mode(const u32 boot_device)
 		return MMCSD_MODE_RAW;
 	}
 }
+#endif
 
 static u32 __get_backup_bootmedia(u32 main_devstat)
 {
@@ -421,13 +422,13 @@ static u32 __get_primary_bootmedia(u32 main_devstat)
 u32 spl_boot_device(void)
 {
 	u32 devstat = readl(CTRLMMR_MAIN_DEVSTAT);
+	u32 bootindex = readl(CONFIG_SYS_K3_BOOT_PARAM_TABLE_INDEX);
 
 	if (bootindex == K3_PRIMARY_BOOTMODE)
 		return __get_primary_bootmedia(devstat);
 	else
 		return __get_backup_bootmedia(devstat);
 }
-#endif
 
 #ifdef CONFIG_SYS_K3_SPL_ATF
 
-- 
2.35.3

